lib/operations/add_user.js

```diff
@@ -1,186 +1,113 @@
 'use strict';
 
 const Aspect = require('./operation').Aspect;
-const defineAspects = require('./operation').defineAspects;
-const OperationBase = require('./operation').OperationBase;
 const CommandOperation = require('./command');
-const applyWriteConcern = require('../utils').applyWriteConcern;
-const CONSTANTS = require('../constants');
-const count = require('./collection_ops').count;
+const defineAspects = require('./operation').defineAspects;
 const crypto = require('crypto');
 const handleCallback = require('../utils').handleCallback;
-const loadDb = require('../dynamic_loaders').loadDb;
-const MongoError = require('../core').MongoError;
-const ReadPreference = require('../core').ReadPreference;
 const toError = require('../utils').toError;
-const updateOne = require('./collection_ops').updateOne;
+const WriteConcern = require('../write_concern');
 
-class AddUserOperation extends OperationBase {
+class AddUserOperation extends CommandOperation {
   constructor(db, username, password, options) {
-    super(options);
+    super(db, {}, options);
```

Comment:
Since you've migrated to using `_buildCommand` I think you no longer need to pass a command to `CommandOperation`s constructor. At very least, the constructor should be able to dynamically determine that a command has been passed in or not, so you can avoid allocating an object here needlessly every time this operation is instantiated.